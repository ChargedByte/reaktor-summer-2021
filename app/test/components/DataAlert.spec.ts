import { createLocalVue, mount } from '@vue/test-utils'

import Vuetify from 'vuetify'

import DataAlert from '~/components/DataAlert.vue'
import AlertObject from '~/model/alertObject'

describe('DataAlert', () => {
  const localVue = createLocalVue()
  let vuetify: Vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  test('is a Vue instance', () => {
    const object = new AlertObject('info', '', false)

    const wrapper = mount(DataAlert, {
      localVue,
      vuetify,
      propsData: { object },
    })
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly: warning', () => {
    const object = new AlertObject('warning', 'Test with warning type', false)

    const wrapper = mount(DataAlert, {
      localVue,
      vuetify,
      propsData: { object },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('renders correctly: warning + reload', () => {
    const object = new AlertObject('warning', 'Test with warning type', true)

    const wrapper = mount(DataAlert, {
      localVue,
      vuetify,
      propsData: { object },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('renders correctly: error', () => {
    const object = new AlertObject('error', 'Test with error type', false)

    const wrapper = mount(DataAlert, {
      localVue,
      vuetify,
      propsData: { object },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('renders correctly: error + reload', () => {
    const object = new AlertObject('error', 'Test with error type', true)

    const wrapper = mount(DataAlert, {
      localVue,
      vuetify,
      propsData: { object },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
