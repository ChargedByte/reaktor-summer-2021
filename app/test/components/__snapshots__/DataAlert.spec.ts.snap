// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataAlert renders correctly: error + reload 1`] = `
"<div role=\\"alert\\" class=\\"v-alert v-sheet theme--dark v-alert--border v-alert--dense v-alert--prominent v-alert--border-left error\\">
  <div class=\\"v-alert__wrapper\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate v-alert__icon mdi mdi-alert theme--dark\\"></i>
    <div class=\\"v-alert__content\\">
      <div class=\\"row align-center\\">
        <div class=\\"grow col\\">
          Test with error type
        </div>
        <div class=\\"shrink col\\"><button type=\\"button\\" class=\\"v-btn v-btn--text theme--dark v-size--default\\"><span class=\\"v-btn__content\\">Reload</span></button></div>
      </div>
    </div>
    <div class=\\"v-alert__border v-alert__border--left\\"></div>
  </div>
</div>"
`;

exports[`DataAlert renders correctly: error 1`] = `
"<div role=\\"alert\\" class=\\"v-alert v-sheet theme--dark v-alert--border v-alert--dense v-alert--border-left error\\">
  <div class=\\"v-alert__wrapper\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate v-alert__icon mdi mdi-alert theme--dark\\"></i>
    <div class=\\"v-alert__content\\">
      Test with error type
    </div>
    <div class=\\"v-alert__border v-alert__border--left\\"></div>
  </div>
</div>"
`;

exports[`DataAlert renders correctly: warning + reload 1`] = `
"<div role=\\"alert\\" class=\\"v-alert v-sheet theme--dark v-alert--border v-alert--dense v-alert--prominent v-alert--border-left warning\\">
  <div class=\\"v-alert__wrapper\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate v-alert__icon mdi mdi-exclamation theme--dark\\"></i>
    <div class=\\"v-alert__content\\">
      <div class=\\"row align-center\\">
        <div class=\\"grow col\\">
          Test with warning type
        </div>
        <div class=\\"shrink col\\"><button type=\\"button\\" class=\\"v-btn v-btn--text theme--dark v-size--default\\"><span class=\\"v-btn__content\\">Reload</span></button></div>
      </div>
    </div>
    <div class=\\"v-alert__border v-alert__border--left\\"></div>
  </div>
</div>"
`;

exports[`DataAlert renders correctly: warning 1`] = `
"<div role=\\"alert\\" class=\\"v-alert v-sheet theme--dark v-alert--border v-alert--dense v-alert--border-left warning\\">
  <div class=\\"v-alert__wrapper\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate v-alert__icon mdi mdi-exclamation theme--dark\\"></i>
    <div class=\\"v-alert__content\\">
      Test with warning type
    </div>
    <div class=\\"v-alert__border v-alert__border--left\\"></div>
  </div>
</div>"
`;
